src/Compilers/CSharp/Portable/Binder/UsingStatementBinder.cs

```diff
@@ -132,9 +144,33 @@ internal override BoundStatement BindUsingStatementParts(DiagnosticBag diagnosti
                 expressionOpt,
                 iDisposableConversion,
                 boundBody,
+                disposeMethod,
                 hasErrors);
         }
 
+        /// <summary>
+        /// Checks for a Dispose method on exprType. Failing to satisfy the pattern is not an error -
+        /// it just means we have to check for an interface instead.
+        /// </summary>
+        /// <param name="exprType">Type of the expression over which to iterate</param>
+        /// <param name="diagnostics">Populated with warnings if there are near misses</param>
+        /// <returns>True if a matching method is found (still need to verify return type).</returns>
+        private MethodSymbol TryFindDisposePatternMethod(TypeSymbol exprType, DiagnosticBag diagnostics)
+        {
+            LookupResult lookupResult = LookupResult.GetInstance();
+            SyntaxNode syntax = _syntax.Expression != null ? (SyntaxNode)_syntax.Expression : (SyntaxNode)_syntax.Declaration;
+            MethodSymbol disposeMethod = FindPatternMethod(exprType, WellKnownMemberNames.DisposeMethodName, lookupResult, syntax, warningsOnly: true, diagnostics, _syntax.SyntaxTree, MessageID.IDS_Disposable);
+            lookupResult.Free();
```

Comment:
This `lookupResult` logic is crazy complicated (not your fault and not something I would necessarily fix in this PR).